pallets/resource-commenting/src/mock.rs

```diff
@@ -0,0 +1,236 @@
+    type BlockNumber = BlockNumber;
+    type Hash = H256;
+    type Hashing = BlakeTwo256;
+    type AccountId = AccountId;
+    type Lookup = IdentityLookup<Self::AccountId>;
+    type Header = Header;
+    type RuntimeEvent = RuntimeEvent;
+    type BlockHashCount = BlockHashCount;
+    type DbWeight = ();
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = pallet_balances::AccountData<Balance>;
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+    type SS58Prefix = SS58Prefix;
+    type OnSetCode = ();
+    type MaxConsumers = frame_support::traits::ConstU32<16>;
+}
+
+parameter_types! {
+    pub const MinimumPeriod: u64 = 5;
+}
+
+impl pallet_timestamp::Config for Test {
+    type Moment = u64;
+    type OnTimestampSet = ();
+    type MinimumPeriod = MinimumPeriod;
+    type WeightInfo = ();
+}
+
+parameter_types! {
+    pub const ExistentialDeposit: u64 = 1;
+}
+
+impl pallet_balances::Config for Test {
+    type Balance = Balance;
+    type DustRemoval = ();
+    type RuntimeEvent = RuntimeEvent;
+    type ExistentialDeposit = ExistentialDeposit;
+    type AccountStore = System;
+    type WeightInfo = ();
+    type MaxLocks = ();
+    type MaxReserves = ();
+    type ReserveIdentifier = ();
+}
+
+impl pallet_permissions::Config for Test {
+    type DefaultSpacePermissions = DefaultSpacePermissions;
+}
+
+impl pallet_space_follows::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type WeightInfo = ();
+}
+
+parameter_types! {
+  pub const MaxCommentDepth: u32 = 10;
+}
+
+impl pallet_posts::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type MaxCommentDepth = MaxCommentDepth;
+    type IsPostBlocked = ();
+    type WeightInfo = ();
+}
+
+pub struct FakeImpls;
+
+impl PermissionChecker for FakeImpls {
+    type AccountId = AccountId;
+
+    fn ensure_user_has_space_permission(
+        user: User<Self::AccountId>,
+        ctx: SpacePermissionsContext,
+        permission: SpacePermission,
+        error: DispatchError,
+    ) -> DispatchResult {
+        Ok(())
+    }
+}
+
+impl SpaceFollowsProvider for FakeImpls {
+    type AccountId = AccountId;
+
+    fn is_space_follower(account: Self::AccountId, space_id: SpaceId) -> bool {
+        false
+    }
+}
+
+parameter_types! {
+    pub const MaxSpacesPerAccount: u32 = 4096;
+}
+
+impl pallet_spaces::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type Roles = FakeImpls;
+    type SpaceFollows = FakeImpls;
+    type IsAccountBlocked = ();
+    type IsContentBlocked = ();
+    type MaxSpacesPerAccount = MaxSpacesPerAccount;
+    type WeightInfo = ();
+}
+
+parameter_types! {
+    pub static ResourcesSpaceId: SpaceId = 0;
+    pub static MaxResourcesIdLength: u32 = 10;
+}
+
+impl pallet_resource_commenting::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type ResourcesSpaceId = ResourcesSpaceId;
+    type MaxResourcesIdLength = MaxResourcesIdLength;
```

Comment:
No. Because in the future, we may want this Id to be the hash of the URI sent from the frontend. If it is hashed, it will not become a URI anymore but a mere opaque ID.